stl/inc/memory

```diff
@@ -1486,6 +1486,8 @@ template <class _Yty, class = void>
 struct _Can_scalar_delete : false_type {};
 template <class _Yty>
 struct _Can_scalar_delete<_Yty, void_t<decltype(delete _STD declval<_Yty*>())>> : true_type {};
+template <>
+struct _Can_scalar_delete<void> : false_type {};

```

Comment:
We need specializations for `const`-, `volatile`-, and `const volatile`-`void` here as well. Or we could drop this change and have the above partial specialization instead derive from negation<is_void<_Yty>>`.